### PR TITLE
refactor(series8): update series8 to use BaseProvider class

### PR DESCRIPTION
**Pull request by ddivad**
## refactor(series8): update series8 to use BaseProvider class

Update Series8 to use the BaseProvider class.

### What's in this PR:
- Refactor the `Series8 ` provider to use the `BaseProvider` class.
- Add logging functionality into `BaseProvider` to be used by adding`this.logger.<level>`
- Add new util functions:
  > removeYearFromTitle: remove year from a show title, ie: title (2019) -> title

  > getClientIp: returns client ip from `req` object in a formatted way for requests.

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -89,5 +89,21 @@ module.exports = {
      */
     timeout: (ms) => {
         return new Promise(resolve => setTimeout(resolve, ms));
+    },
+
+     /**
+     * returns formatted clientId
+     * @param req request sent to server.
+     */
+    getClientIp: (req) => {
```
Comment (FlyPigsFly) [inline src/utils/index.js]: Question. Does getClientIp need to be added to this file when it already exists at https://github.com/ApolloTVofficial/Claws/blob/117a3ffb998f268dc8aab8850c1342632d14ba7f/src/scrapers/providers/BaseProvider.js#L88 Seems redundant to be located in multiple places.

Comment (ddivad) [reply src/utils/index.js]: No, you're right! I missed that function in `BaseProvider` actually - I'll fix this evening.

Comment (ddivad) [reply src/utils/index.js]: fixed @FlyPigsFly :)

File: src/scrapers/providers/tv/Series8.js [hunk 2]
```diff
@@ -81,29 +56,18 @@ async function Series8(req, sse) {
...
-            videoUrls.forEach(async (provider) => {
+            videoUrls.forEach(async (link) => {
                 const headers = {
-                    'user-agent': userAgent,
+                    'user-agent': randomUseragent.getRandom(),
                     'x-real-ip': clientIp,
                     'x-forwarded-for': clientIp
                 };
-                resolvePromises.push(resolve(sse, provider, 'Series8', jar, headers));
+                resolvePromises.push(this.resolveLink(link, ws, jar, headers));
```
Comment (NBTX) [inline src/scrapers/providers/tv/Series8.js]: Is the name pushed anywhere else or will this end up returning 'null' for the provider?

Comment (ddivad) [reply src/scrapers/providers/tv/Series8.js]: It's done in BaseProvider so it should be ok

Comment (NBTX) [reply src/scrapers/providers/tv/Series8.js]: Nevermind, I see the provider name is handled in the BaseProvider class.

**Pull request merged**